### PR TITLE
Terraform provisioning basic elastic scalable infrastructure

### PR DESCRIPTION
**Pull request by boymaas**
This is the base on which we can add more intelligent health checks, and the introduction of a snapshot node.

### CODE REVIEW

File: infra/net/variables.tf [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+
+variable "sg_name"{
+ type = string
+ default = "alb_sg"
+}
+
+variable "sg_description"{
+ type = string
+ default = "SG for application load balancer"
```
Comment (shelbyd) [inline infra/net/variables.tf:57]: Not sure what SG means. If it's an acronym, please use the full version for extra clarity.

File: blockchain/node/src/chain_spec.rs [hunk 2]
```diff
@@ -109,7 +116,11 @@ pub fn local_testnet_config() -> Result<ChainSpec, String> {
                     get_account_id_from_seed::<sr25519::Public>("Eve//stash"),
                     get_account_id_from_seed::<sr25519::Public>("Ferdie//stash"),
                 ],
-                true,
+                FractalMintingConfig {
+                    fractal_authoritative_account: get_account_id_from_seed::<sr25519::Public>(
```
Comment (shelbyd) [inline blockchain/node/src/chain_spec.rs:120]: I think the type for get_account_id_from_seed will be inferred. Feel free to keep for explicitness.

**Review by shelbyd — CHANGES_REQUESTED** [review]

File: infra/net/variables.tf (hunk 1 above)
Comment (melatron) [reply infra/net/variables.tf:57]: SG - security group in AWS. The acronym is quite used this way, but we can change it.

File: blockchain/node/src/chain_spec.rs [hunk 3]
```diff
@@ -160,8 +171,102 @@ fn testnet_genesis(
...
+                            .unchecked_into(),
+                        // 5FCLgr3aRRHCJLXeoH5vbz7NGuQ3E3hLg44KPn4LaRMb7ub9
+                        hex!["8a8766b57f2391f4f6e2fed09b1ea8fde72948f988494ec39e91562812c5dee5"]
+                            .unchecked_into(),
+                    ),
+                ],
+                // Sudo account
+                // 5FCLsu6FTwi9cafruqVbEo8uUEqPHaf6oqNLhSTn5CkbrQ3o
+                hex!["8a880aedd15c00bfa62220f014d12ff818534e99bd298c5c1a68ac221fc2471e"].into(),
+                // 5FCLidfiL1wcTSXvecm1PrrP3N3jUxAgpdDGJQRAqA5pk1K3
```
[the remark this answers is not in the document]
Comment (melatron) [reply blockchain/node/src/chain_spec.rs:257]: This account is provided for the `FractalMintingConfig`, don't we need it? From how I understand it we used the similar config for the testnet.

[the remark this answers is not in the document]
Comment (shelbyd) [reply blockchain/node/src/chain_spec.rs:259]: I was saying to add a comment above it like we did for the Sudo account.

File: infra/net/variables.tf (hunk 1 above)
Comment (shelbyd) [reply infra/net/variables.tf:57]: That's fine if it's just my lack of experience meaning I don't know it.

**Review by shelbyd — COMMENTED** [review]
Please reply to / resolve review comments that you address, so the reviewer doesn't have to figure out what which you've addressed. A simple "Done" is great.

File: blockchain/node/src/chain_spec.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (melatron) [reply blockchain/node/src/chain_spec.rs:260]: Done

File: infra/net/main.tf [hunk 4]
```diff
@@ -0,0 +1,457 @@
...
+  availability_zone       = "${var.region}a"
+  map_public_ip_on_launch = true
+  tags = {
+    Project = "fclnet-provisioning"
+     Name = "public_subnet1"
+
+ }
+}
+
+# Create Public Subnet2
```
[the remark this answers is not in the document]
Comment (melatron) [reply infra/net/main.tf:40]: @boymaas would be more suited to answer this.

[the remark this answers is not in the document]
Comment (boymaas) [reply infra/net/main.tf:40]: This is correct, two availability zones leads to higher resilience. I created public and private to give us an option to let the nodes be approachable from the outside for development/testing, and if it is a business requirement to move them into the private network its a trivial change to the config. If this is not needed, we can remove the private subnet(s)

**Review by melatron — APPROVED** [review]

File: infra/net/main.tf (hunk 4 above)
[the remark this answers is not in the document]
Comment (shelbyd) [reply infra/net/main.tf:40]: Might be good to include as a comment somewhere in this file. Could be a quick sentence with a link to AWS docs on it.

**Review by shelbyd — APPROVED** [review]

**Pull request merged**